testsuite/MDAnalysisTests/lib/test_nsgrid.py

```diff
@@ -231,3 +231,14 @@ def test_nsgrid_probe_close_to_box_boundary():
     expected_dists = np.array([2.3689647], dtype=np.float64)
     assert_equal(results.get_pairs(), expected_pairs)
     assert_allclose(results.get_pair_distances(), expected_dists, rtol=1.e-6)
+
+
+def test_zero_max_dist():
```

Comment:
This previously used to segfault (see #2656)

Reply:
Could you add a comparison to the expected value of no pairs found?